controllers/usersController.js

```diff
@@ -262,6 +262,46 @@ class UserController {
       });
     }
   }
+
+  static async changePassword(req, res) {
+    try {
+      const {
+        body: { newPassword, currentPassword },
+        user: { username: usernameFromToken },
+        params: { username }
+      } = req;
+      const salt = await genSaltSync(
+        parseFloat(process.env.BCRYPT_HASH_ROUNDS) || 10
+      );
+      const response = await User.findOne({ where: { username } });
+
+      const newHashedPassword = await hashSync(newPassword, salt);
+
+      if (!response) {
+        return res.status(NOT_FOUND).json({ message: "User not found" });
+      }
+      if (usernameFromToken !== username) {
+        return res.status(UNAUTHORIZED).json({ error: "Not authorized" });
+      }
+      const newPWdMatchCurrent = await compareSync(
+        currentPassword,
+        response.password
+      );
+      if (!newPWdMatchCurrent) {
+        return res.status(BAD_REQUEST).json({
+          message: "Invalid Current Password"
+        });
+      }
+      await User.update({ password: newHashedPassword }, { where: { username } });
+      return res.status(OK).json({ message: "Password updated successfully" });
+    } catch (error) {
+      res.status(INTERNAL_SERVER_ERROR).json({
+        message: "Password update failed",
```

Comment:
I can't see any difference between `error` and `message` because both are the error messages. Why can't you keep one of them?